### PR TITLE
add tests and update c-index calc

### PR DESCRIPTION
**Pull request by GabrielGimenez**

### CODE REVIEW

File: xgbse/metrics.py [hunk 1]
```diff
@@ -65,28 +65,41 @@ def concordance_index(y_true, survival, risk_strategy="mean", which_window=None)
...
-    for _, row in cind_df.query("e == True").iterrows():
-
-        # getting all censored and uncensored samples
-        # after current row
-        samples_after_i = cind_df.query(f"""{row['t']} < t""")
+    for i in range(all_samples):
+        if events[i]:
+            # getting all censored and uncensored samples
+            # after current row
+            pairs, concord, tied = _concordant_pairs(hazards[i + 1 :], hazards[i])
```
Comment (davivieirab) [inline xgbse/metrics.py:85]: For the last loop: 
when we get up to hazards[i+1 :], if we get a last element as a True event, do we still have an element with index i+1? Doesn't it raise an error?

File: tests/test_extrapolation.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# generating Kaplan Meier for all tests
+
+time_bins = get_time_bins(T_train, E_train, 100)
+
+mean, high, low = calculate_kaplan_vectorized(
+    T_train.values.reshape(1, -1), E_train.values.reshape(1, -1), time_bins
+)
+
+km_survival = pd.concat([mean] * len(y_train))
+km_survival = km_survival.reset_index(drop=True)
```
Comment (davivieirab) [inline tests/test_extrapolation.py:35]: Do we use this dataframe anywhere in the tests file? If not, shouldn't we remove it?

**Review by davivieirab — COMMENTED** [review]
Two comments:
1 - in the test_extrapolation file, do we use the km_survival dataframe anywhere? Is it necessary to be there? If we are testing the transformation (concat, reset_index, etc), shouldn't we be more explicit?

2 - In the xgbse/metrics.py file - concordance_index function:
don't we have an IndexError trying to access the hazards[i+1:] element of this array?

File: xgbse/metrics.py (hunk 1 above)
Comment (GabrielGimenez) [reply xgbse/metrics.py:85]: It doesn't because it's a slice, it'll just be an empty list

File: tests/test_extrapolation.py (hunk 2 above)
Comment (GabrielGimenez) [reply tests/test_extrapolation.py:35]: We use it in line 81, to assert that c-index of km is 0.5

File: xgbse/metrics.py (hunk 1 above)
Comment (davivieirab) [reply xgbse/metrics.py:85]: That's right! Nice point

File: tests/test_extrapolation.py (hunk 2 above)
Comment (davivieirab) [reply tests/test_extrapolation.py:35]: Is it in this .py file? test_extrapolation.py goes up to line 69..

Comment (davivieirab) [reply tests/test_extrapolation.py:35]: You're talking about test_metrics.py file, right? I don't see in this file any usage for km_survival.

Comment (GabrielGimenez) [reply tests/test_extrapolation.py:35]: you're right, we don't use it here, will remove

**Pull request merged**